### DISCUSSION TITLE
How to split read/write queries to Slave/Master datasources

### DISCUSSION THREAD

**Discussion by rostamiani** [prisma/prisma#13855] [Q&A]
TL;DR
I have one master postgerSql DB with a readonly replication server. I need to send all FindFirst, FindMany, Count,... queries to the replica Address and others to the master

I'm using Prisma 3.15 with NestJS and need to split queries to two different data-sources. One is read-only replica and another is master database.
### I don't want to manage these manually on each prisma call!
How can I split them automatically? Any ideas?

I can define two separate clients in PrismaService like this:
```ts
@Injectable()
export class PrismaService implements OnModuleInit {
  constructor(
    private readClient: PrismaClient,
    private writeClient: PrismaClient
  ) {
    this.readClient = new PrismaClient({
      datasources: {db: {url: process.env.PRISMA_READ_DB_URL}},
    });
    this.writeClient = new PrismaClient({
      datasources: {db: {url: process.env.PRISMA_WRITE_DB_URL}},
    });
  }
```
#####  But how can I split queries between them?

Can I override prisma modules like this?
```ts
  get user() {
    // Do something with the user!!!

    return this.readClient.user;
  }
```

Thanks

**Comment by austincrim** [comment]
Hey @rostamiani 👋,

Could you give me more details about "splitting queries between them"? Do you mean some kind of load balancing technique?

**Reply by rostamiani** [reply]
I have one master postgerSql DB with a readonly replication server. I need to send all FindFirst, FindMany, Count,... queries to the replica Address and others to the master

**Reply by austincrim** [reply]
There isn't a built-in way to do this in Prisma today. You will need to write an abstraction layer above the two clients or manage it manually for now.

Feel free to take a look through [this relevant issue](https://github.com/prisma/prisma/issues/7161) and comment your thoughts on how we can consider your use case when designing the feature.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
